src/io/flutter/jxbrowser/EmbeddedBrowser.java

```diff
@@ -81,6 +85,13 @@ public void projectClosing(@NotNull Project project) {
     });
   }
 
+  /**
+   * This is to clear out a potentially old URL, i.e. a URL from an app that's no longer running.
+   */
+  public void resetUrl() {
+    this.mainUrlFuture = new CompletableFuture<>();
```

Comment:
Replacing a completable future with another like this has lead to bugs in the past. A reasonable pattern is to only reset the future if it has not yet completed or to complete the existing future with a failure value depending on what you want to do. Otherwise you can end up with strange behavior if someone was waiting on the future you reset.

Reply:
Okay, we do want to replace the future even if it has completed successfully (since a new URL will have a different vm service URL), but it would make sense to complete with failure if it's not done yet.

Reply:
completing with null since this is a race condition instead of an error we care about.